library/auth/src/main/java/com/epmedu/animeal/auth/AuthAPI.kt

```diff
@@ -1,152 +1,37 @@
 package com.epmedu.animeal.auth
 
-import com.amplifyframework.auth.AuthUserAttribute
-import com.amplifyframework.auth.AuthUserAttributeKey
-import com.amplifyframework.auth.cognito.AWSCognitoAuthSession
-import com.amplifyframework.auth.cognito.options.AWSCognitoAuthSignInOptions
-import com.amplifyframework.auth.cognito.options.AuthFlowType
-import com.amplifyframework.auth.options.AuthSignUpOptions
-import com.amplifyframework.auth.result.AuthSessionResult
-import com.amplifyframework.core.Amplify
-import com.epmedu.animeal.extensions.suspendCancellableCoroutine
-import kotlin.coroutines.resume
-import kotlin.coroutines.resumeWithException
+@Suppress("ComplexInterface")
+interface AuthAPI {
 
-@Suppress("TooManyFunctions")
-class AuthAPI {
-
-    var authenticationType: AuthenticationType = AuthenticationType.Mobile
-        private set
-
-    private val AWSCognitoAuthSession.isSignedInWithoutErrors
-        get() = isSignedIn &&
-            awsCredentialsResult.type == AuthSessionResult.Type.SUCCESS &&
-            identityIdResult.type == AuthSessionResult.Type.SUCCESS &&
-            userPoolTokensResult.type == AuthSessionResult.Type.SUCCESS &&
-            userSubResult.type == AuthSessionResult.Type.SUCCESS
-
-    suspend fun getCurrentUserId(): String = suspendCancellableCoroutine {
-        Amplify.Auth.getCurrentUser(
-            { user ->
-                resume(user.username)
-            },
-            { authException ->
-                resumeWithException(authException)
-            }
-        )
-    }
-
-    suspend fun isSignedIn(): Boolean {
-        return suspendCancellableCoroutine {
-            Amplify.Auth.fetchAuthSession(
-                { session ->
-                    resume(
-                        when (session) {
-                            is AWSCognitoAuthSession -> session.isSignedInWithoutErrors
-                            else -> session.isSignedIn
-                        }
-                    )
-                },
-                {
-                    resume(false)
-                }
-            )
-        }
-    }
-
-    fun setMobileAuthenticationType() {
-        authenticationType = AuthenticationType.Mobile
-    }
-
-    fun setFacebookAuthenticationType(isPhoneNumberVerified: Boolean) {
-        authenticationType = AuthenticationType.Facebook(isPhoneNumberVerified)
-    }
+    var authenticationType: AuthenticationType
 
+    suspend fun getCurrentUserId(): String
+    suspend fun isSignedIn(): Boolean
+    fun setMobileAuthenticationType()
+    fun setFacebookAuthenticationType(isPhoneNumberVerified: Boolean)
```

Comment:
I guess these two functions are redundant now. Aren't they?

Reply:
correct, I removed them as well. Thanks for pointing out.